examples/wire_read_tcp.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python
+import argparse
+
+import dns.message
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Read sequence of DNS wire formats prefixed with 2-byte "
+        "length field - like from TCP socket - and print the messages. This "
+        "format is used e.g. by dnsperf -B option and dnsgen."
+    )
+    parser.add_argument("infile")
+    args = parser.parse_args()
+
+    with open(args.infile, "rb") as infile:
+        while True:
+            len_wire = infile.read(2)
+            if len(len_wire) == 0:
+                break
```

Comment:
Should there be a check for partial read?  If there's a single byte at the end, this will read it, successfully (but incorrectly) call `int.from_bytes` on the single byte to parse a length.  Then, it will either fail to read the bytes of the message, or if the single byte was 0, will successfully read 0 bytes and fail to parse the message.